packages/contracts/deploy/new/10_framework/01_ens_subdomains.ts

```diff
@@ -1,13 +1,54 @@
-import {DeployFunction} from 'hardhat-deploy/types';
-import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {ENSRegistry, ENSRegistry__factory} from '../../../typechain';
 import {
   getContractAddress,
   getENSAddress,
   getPublicResolverAddress,
   registerSubnodeRecord,
   transferSubnodeChain,
 } from '../../helpers';
-import {ENSRegistry__factory} from '../../../typechain';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+import {DeployFunction} from 'hardhat-deploy/types';
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+
+async function registerAndTransferDomain(
+  ensRegistryContract: ENSRegistry,
+  managingDAOAddress: string,
+  domain: string,
+  node: string,
+  deployer: SignerWithAddress,
+  hre: HardhatRuntimeEnvironment,
+  ethers: any
+) {
+  let owner = await ensRegistryContract.owner(node);
+
+  // node hasn't been registered yet
+  if (owner === ethers.constants.AddressZero) {
+    owner = await registerSubnodeRecord(
+      domain,
+      deployer,
+      await getENSAddress(hre),
+      await getPublicResolverAddress(hre)
+    );
+  }
+
+  if (owner !== managingDAOAddress && owner !== deployer.address) {
+    throw new Error(
+      `${domain} is not owned either by deployer: ${deployer.address} or management dao: ${managingDAOAddress}. 
+      Check if the domain is owned by ENS wrapper and if so, unwrap it from the ENS app.`
```

Comment:
Who is "ENS wrapper" and what do you mean by "unwrap it from the ENS app"?

Reply:
ENS Wrapper is a new kind of solution ENS does. [See](https://metaversal.banklesshq.com/p/name-wrapper)